Sources/GRPCHTTP2Core/GRPCMessageFramer.swift

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import NIOCore
+
+/// A ``GRPCMessageFramer`` helps with the framing of gRPC data frames:
+/// - It prepends data with the required metadata (compression flag and message length).
+/// - It compresses messages using the specified compression algorithm (if configured).
+/// - It coalesces multiple messages (appended into the `Framer` by calling ``append(_:compress:)``)
+/// into a single `ByteBuffer`.
+struct GRPCMessageFramer {
+  /// Length of the gRPC message header (1 compression byte, 4 bytes for the length).
+  static let metadataLength = 5
+
+  /// Maximum size the `writeBuffer` can be when concatenating multiple frames.
+  /// This limit will not be considered if only a single message/frame is written into the buffer, meaning
+  /// frames with messages over 64KB can still be written.
+  /// - Note: This is expressed as the power of 2 closer to 6KB (i.e., 6KiB) because `ByteBuffer`
+  /// reserves capacity in powers of 2. This way, we can take advantage of the whole buffer.
+  static let maximumWriteBufferLength = 16_384
```

Comment:
I'm a bit confused by the numbers here, the max length is 16 KiB but the docs talk about 64 KiB and 6KiB, can you clear this up?